From 2e156af7cf49ed50e3f6f010cd12d1004f609369 Mon Sep 17 00:00:00 2001
From: Luwei Zhou <b45643@freescale.com>
Date: Wed, 30 Oct 2013 16:24:06 +0800
Subject: [PATCH 424/505] ENGR00285516 linux-test: mxc_mlb_test: Fix the
 sending issue of ISOC mode

This patch provides a workaround for transmitting issue of MLB ISOC mode.
In the test case of ISOC mode, MITB recieves less than 5000 packets but
Sabreauto side has invoked sending API for 5000 times. Since there is
no description in MLB150 for transmission complete flag. So we just
use this workaround to fix this issue. After adding the 1ms sleep,
both side can pass the test.

Signed-off-by: Luwei Zhou <b45643@freescale.com>
---
 test/mxc_mlb_test/mxc_mlb150_test.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/mxc_mlb_test/mxc_mlb150_test.c b/test/mxc_mlb_test/mxc_mlb150_test.c
index 43c7374..a24b936 100644
--- a/test/mxc_mlb_test/mxc_mlb150_test.c
+++ b/test/mxc_mlb_test/mxc_mlb150_test.c
@@ -376,7 +376,7 @@ int do_txrx_test(int fd)
 	}
 
 shutdown:
-
+	usleep(1000);
 	ret = ioctl(fd, MLB_CHAN_SHUTDOWN);
 	if (ret) {
 		printf("Failed to shutdown async channel\n");
-- 
1.8.0

